### ISSUE TITLE
[Due for payment 2025-04-02] [Due for payment 2025-03-28] [Due for payment 2025-02-18] [Native Share] Implement Native Share for IOS

### ISSUE THREAD

**Issue by grgia** [Expensify/App#48788]
## Description 
This is a tracking issue for the IOS implementation of the Native Share project. This section of the project will be handled separately but merged alongside [the Android issue](https://github.com/Expensify/App/issues/48789).

## Project 
https://docs.google.com/document/d/14C1VifxvIXeyLAf0XstbC-wvyAb5SQZ0AOw0-jLRyZs/edit

## Tracking  
https://github.com/Expensify/App/issues/36613

<details><summary>Issue Owner</summary>Current Issue Owner: @sonialiap</details>

**Comment by BrtqKr** [comment]
Hey, I'm from SWM, I'd like to take care of this issue.

**Comment by grgia** [comment]
Bumped for update / steps for unblocking permissions

**Comment by BrtqKr** [comment]
I am focusing on split navigation atm. I'll try to check the setup whenever it's applied and try to continue whenever the navigation is ready.

**Comment by BrtqKr** [comment]
Taking care of this atm. I've checked the groups before the weekend and they seemed to work as expected.

**Comment by BrtqKr** [comment]
Working on the files processing - they are being saved correctly from the share extenbsion, but there's something wrong with retrieving them on the react native side. This might be something related to the permissions, but I'll try to confirm it tomorrow.

**Comment by grgia** [comment]
Thanks @BrtqKr- do you have a draft PR up for your branch?

**Comment by BrtqKr** [comment]
Not yet, wired it up though and the files are being passed to the react native wihthout any issues, so happily there's no need for the additional set-ups. I'll be verifying the scanning/sending with the passed files tomorrow, so this would probably be the initial version for the ios. I'll share more details next evening.

**Comment by BrtqKr** [comment]
Moving forward with that - wired up most of the js parts - I'll continue working on that and the files cleanup after sharing tomorrow

**Comment by grgia** [comment]
Sounds great, thanks @BrtqKr

**Comment by BrtqKr** [comment]
Still working on the JS part - today it was mostly persistence and onyx-related things, which are necessary for removing files - it's not done yet though

**Comment by BrtqKr** [comment]
Continued the topic from yesterday. I've found some edge cases about the cleanup part - there's an issue when the share attempt gets cancelled and the file isn't removed properly, but we can handle this by performing cleanups for the files older than a week. Besides I had to slightly modify the model for the image, but apply small updates to the doc when I'm sure everything works.

**Comment by BrtqKr** [comment]
@grgia, I've finished wiring up the core version for the ios today. Also, @filip-solecki mentioned that there wasn't much left regarding the Android native part, so I'd say that the remaining things before the PR would be:

1. Adjust the data format on Android to match the ios (I had to slightly change the format to handle some cases on the react-native side)
2. Finish the file removal
3. Adjust the display for the files
4. Pass to @staszekscp and @war-in to apply the hybrid app-related changes and test it properly

The first three points shouldn't take much time since those are just the follow-ups to the already existing parts (the second point is a bit troublesome to test though). I'm not sure about the fourth point. As long as everything goes as expected on the hybrid part we should be able to send the final version sometime at the beginning the next week, but this is something @staszekscp or @war-in would need to estimate when they start applying the changes.

https://github.com/user-attachments/assets/5c5bfd4a-4b0e-4537-94d5-cefc8d6b3cca

**Comment by filip-solecki** [comment]
Hi! Quick update - I took the issue from @BrtqKr and I'll work on both right now as JS part will be shared for both Android and iOS

**Comment by grgia** [comment]
@BrtqKr This update looks great, thank you! Cool to see this start to take form

**Comment by grgia** [comment]
> Hi! Quick update - I took the issue from @BrtqKr and I'll work on both right now as JS part will be shared for both Android and iOS

@filip-solecki great, could you give me an update of where you're at with this for the week?

**Comment by grgia** [comment]
Also @filip-solecki (cc @BrtqKr ) do you have a rough ETA for when we can get the PR up for these?

**Comment by filip-solecki** [comment]
Hi! Still working on Submit page which is a little bit tricky as it combines a lot of logic. I am going to be OOO from tomorrow till the end of the week, so the realistic ETA for PR is Wednesday/Thursday of the next week, maybe earlier(Monday/Tuesday) if there will be someone available to continue current work

**Comment by cdOut** [comment]
Hi, since Filip is going to be OOO for the rest of this week I'll be taking over the share flow tasks for that timeframe.

**Comment by cdOut** [comment]
Hey, quick update. I've got into the project and finished the Submit flow for IOS. I've also fixed the issue with the images sometimes not displaying properly on the receipt page. I'm now looking into merging the Android branch for the native share flow into the IOS one to consolidate it into one.

**Comment by grgia** [comment]
@cdOut @filip-solecki Could you post a current update?

**Comment by filip-solecki** [comment]
Sure, @cdOut is fixing image displaying issues which happen sometimes and he also finished Submit flow so we have to finish Android and merge it here and we can push it to review and tests

**Comment by grgia** [comment]
@filip-solecki cc @cdOut mind posting a quick update here as well?

**Comment by grgia** [comment]
Great thank you! @cdOut

**Comment by grgia** [comment]
@cdOut do you have an EOW update?

**Comment by cdOut** [comment]
@grgia I was just a side helper on this issue during Filip's OOO, if any updates on the state of it are in order you should ping @filip-solecki as he's more informed on the state of it and is the main person behind the feature.

**Comment by filip-solecki** [comment]
Hi @grgia, so on the iOS side most of the issues have been resolved, now we need to block sharing for unsupported file formats and address some minor JS issues

**Comment by grgia** [comment]
Great @filip-solecki could you give me an update on when you think we could get a draft PR up?

**Comment by filip-solecki** [comment]
Hi! So I think that the realistic date may be Wednesday/Thursday in the next week

**Comment by grgia** [comment]
Great, keep me posted @filip-solecki

**Comment by filip-solecki** [comment]
Hi @grgia ! Unfortunately I was sick most of this week and we won't be able to prepare PR today, it should be ready in the mid of next week I hope

**Comment by grgia** [comment]
@filip-solecki sorry to hear that, hope you're feeling better! How's this week for you?

**Comment by filip-solecki** [comment]
This is much better - both with health and this issue, I've already finished Submit flow process, it just needs a little polishing while @Guccio163 is working on Attachment Preview and styling all pages across the flow. We also have to make sure we won't let user go back to sharing process after submitting message or transaction and block unsupported files sharing on iOS which is a little bit tricky moreover there are a few smaller polishes that we have to work on

**Comment by grgia** [comment]
> there are a few smaller polishes that we have to work on

@filip-solecki great update! I tagged the design team to get this answered. Could you confirm if there's anything else blocking from getting the PR up? Could we tackle polish type issues as follow ups?

**Comment by grgia** [comment]
Hey @filip-solecki, could we get a draft PR up for this?

**Comment by grgia** [comment]
Hey @filip-solecki could we go back to daily updates?

**Comment by filip-solecki** [comment]
Hi @grgia ! Of course, So today @Guccio163 was working on displaying AttachmentModal when clicking on thumbnail in Share flow, while I was fixing PDF uploading and investigating possible solutions for sharing aspectRatio for videos from native module, we are really close to prepare Draft PR and allow you test everything

**Comment by Guccio163** [comment]
Hi, implementing AttachmentModal logic is going smoothly; I've stumbled upon a problem with handling PDF thumbnail on android and @filip-solecki still have to figure out a couple of native issues, but it looks like it's on a good way to PR soon 👌

**Comment by grgia** [comment]
@Guccio163 how's that going? Can we get a PR up this week or next?

**Comment by filip-solecki** [comment]
Hi! @Guccio163 is working on fixes suggested by design team and I am working on Submit flow for not existing user and these are our latest bigger issues, it is hard to say the exact day, we hope we will be ready tomorrow to prepare draft PR, but it may be also early next week

**Comment by filip-solecki** [comment]
In the meantime I got access to Hybrid app today and I'll try to connect the new Share flow there

**Comment by filip-solecki** [comment]
Hi! 
Today's update is that we didn't succeded with preparing draft PR, we are still working on submit flow to unknown user, also I was checking how we can connect our new share flow with the old one in Hybrid App

**Comment by grgia** [comment]
@filip-solecki how's the hybrid app flow going?

**Comment by filip-solecki** [comment]
Hi! I am working on this, I have to merge modules for processing files cause they are a little bit different as we support other types than only images right now in ND and then I'll have to find the best place to redirect users to ND if they are using it or to stay in OD and process files here. Additionally I have one concern about supported media types. In OD we supported only images and it was easy job. Right now we have a few more media types that we can share to chats in ND. Android allows us to specify exact type of data we want allow user to share, but iOS groups it in specified groups like: images, videos, text, files and attachments. So my questions are:
1. What should we do if someone is not using ND and tries to share video. In OD we don't support it, should we redirect user to ND despite all or display toast message in OD that this type of file is not supported?
2. We are also not able to block e.g. CSV files sharing (AFAIK this type was not in scope) on iOS as it is part of `NSExtensionActivationSupportsFileWithMaxCount` group. Should we then process it where it is possible? Or block and display some generic message that this type is not supported? (Allowing users to share shouldn't require any additional work)

**Comment by grgia** [comment]
> What should we do if someone is not using ND and tries to share video. In OD we don't support it, should we redirect user to ND despite all or display toast message in OD that this type of file is not supported?

@filip-solecki by this do you mean that they don't have the ND app?

>  We are also not able to block e.g. CSV files sharing (AFAIK this type was not in scope) on iOS as it is part of NSExtensionActivationSupportsFileWithMaxCount group. Should we then process it where it is possible? Or block and display some generic message that this type is not supported? (Allowing users to share shouldn't require any additional work)
> AFAIK this type was not in scope

given it's not in scope, we can likely just block CSV file sharing and show users a message that such files are not supported

@JmillsExpensify could you confirm this

**Comment by filip-solecki** [comment]
> @filip-solecki by this do you mean that they don't have the ND app?

I mean they have Hybrid App installed, but never used ND, only working on OD

**Comment by filip-solecki** [comment]
Hi! Quick update:
I am still working on merging methods for files processing as it is much more complicated than I thought before seeing OD method. And @Guccio163 finished navigation issues and will be working on adding message for users while sharing unsupported media type

**Comment by grgia** [comment]
@filip-solecki do you have a rough estimate on ETA?

**Comment by filip-solecki** [comment]
[Here](https://github.com/Expensify/App/pull/54354) is the draft PR for both iOS and Android! We are still working on some really minor polishes, but it would be good if you can test it in your free time and send us some feedback. Today we also want to send it to our internal review  🚀

**Comment by grgia** [comment]
Wooo! Great to see the draft PR 🎉

**Comment by grgia** [comment]
PR in review, adding label!

**Comment by grgia** [comment]
Was OOO, not overdue melvin

**Comment by sonialiap** [comment]
No payment needed, all three engineers are from SWM

**Comment by sonialiap** [comment]
@grgia are we doing more work on this issue?

**Comment by grgia** [comment]
In regression period

**Comment by sonialiap** [comment]
@grgia I see this got linked to a few deploy blockers, did this cause any regressions?

**Comment by grgia** [comment]
@sonialiap no regressions!

**Comment by sonialiap** [comment]
Thanks!

**Comment by sonialiap** [comment]
Payment summary:
- All three engineers are from SWM so no payment needed (I trusted melvin about cdOut needing a manual offer, but was lead astray 😂)

Since this is a new feature, most of the checklist is not relevant

**Comment by grgia** [comment]
@sonialiap I think we forgot to handle the payout for @shubham1206agra's C+ review. Would you please help with $500 review payment for this PR? https://github.com/Expensify/App/pull/58834

**Comment by sonialiap** [comment]
Thanks for reopening!

Payment summary:
- @shubham1206agra for C+ review $500, please request in New Dot

**Comment by garrettmknight** [comment]
$500 approved for @shubham1206agra
